specfile/context_management.py

```diff
@@ -0,0 +1,140 @@
+# Copyright Contributors to the Packit project.
+# SPDX-License-Identifier: MIT
+
+import collections
+import contextlib
+import io
+import os
+import pickle
+import sys
+import tempfile
+import types
+from typing import Any, Callable, Dict, Generator, List, Optional, overload
+
+
+@contextlib.contextmanager
+def capture_stderr() -> Generator[List[bytes], None, None]:
+    """
+    Context manager for capturing output to stderr. A stderr output of anything run
+    in its context will be captured in the target variable of the with statement.
+
+    Yields:
+        List of captured lines.
+    """
+    fileno = sys.__stderr__.fileno()
+    with tempfile.TemporaryFile() as stderr, os.fdopen(os.dup(fileno)) as backup:
+        sys.stderr.flush()
+        os.dup2(stderr.fileno(), fileno)
+        data: List[bytes] = []
+        try:
+            yield data
+        finally:
+            sys.stderr.flush()
+            os.dup2(backup.fileno(), fileno)
+            stderr.flush()
+            stderr.seek(0, io.SEEK_SET)
+            data.extend(stderr.readlines())
+
+
+class GeneratorContextManager(contextlib._GeneratorContextManager):
+    """
+    Extended contextlib._GeneratorContextManager that provides get() method.
+    """
+
+    def __init__(self, function: Callable) -> None:
+        super().__init__(function, tuple(), {})
+
+    def __del__(self) -> None:
+        # make sure the generator is fully consumed, as it is possible
+        # that neither __enter__() nor get() have been called
+        collections.deque(self.gen, maxlen=0)
+
+    def get(self) -> Any:
```

Comment:
Out of the options that you mentioned, I probably like the explicit method the most, although I am not sure about the name. While `get()` is short, it is not very clear what it does (someone could expect it to be get method of dict). What about `.content()`?

Reply:
Property also looks good (imho property/method are clearer than callable or iterable). In the end, it's a matter of taste :) Have you tried taking a look if some popular libraries do something similar so that we could get inspired there? That would probably be ideal.

Reply:
> Have you tried taking a look if some popular libraries do something similar so that we could get inspired there?

Not really, do you know any that would employ this concept?

Reply:
I first though of `requests` since they use some context managers but haven't found anything similar. Let's stick with `content` if that sounds good to you.

Reply:
Let's go with `content` property, we can rename it later.